### DISCUSSION TITLE
Using a tree-sitter parser

### DISCUSSION THREAD

**Discussion by tohoyn** [atom/atom#23565] [Q&A]
I'm writing my own grammar that uses package tree-sitter-sexp. How do I install package tree-sitter-sexp so that atom can find it? Or do I have to put some require statement in the .json file in subdirectory grammar?

**Comment by claytonrcarter** [comment]
You have to install the tree-sitter package into the language grammar package that you're working on, and then Atom will it pick up from there.

Within your language/grammar package, you'll do `npm install --save tree-sitter-sexp`. Then, in your grammar file (probably `grammars/tree-sitter-sexp.cson` or similar) you'll add `type: 'tree-sitter'` and `parser: 'tree-sitter-sexp'` to tell Atom to use tree-sitter and which tree-sitter parser to use.

**Comment by tohoyn** [comment]
ke, 2022-02-09 kello 08:58 -0800, claytonrcarter kirjoitti:
> You have to install the tree-sitter package into the language grammar
> package that you're working on, and then Atom will it pick up from
> there.
> Within your language/grammar package, you'll do npm install --save
> tree-sitter-sexp. Then, in your grammar file (probably grammars/tree-
> sitter-sexp.cson or similar) you'll add type: 'tree-sitter' and
> parser: 'tree-sitter-sexp' to tell Atom to use tree-sitter and which
> tree-sitter parser to use.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android.
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>

Now I get the following error:

The module '/home/tohoyn/github/tree-sitter-
sexp/build/Release/tree_sitter_sexp_binding.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 102. This version of Node.js requires
NODE_MODULE_VERSION 80. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`). in
/home/tohoyn/.atom/packages/language-test1/grammars/test1.cson

I have tried "npm rebuild" and "npm install" but they don't help.

     - Tommi

**Reply by claytonrcarter** [reply]
Is that in the Atom console, or on the CLI? As I understand it, this is b/c you have used 1 version of `node` to compile your tree-sitter code, but Atom/electron is using a different version of `node` internally. You might try running `npx electron-rebuild --version 9.4.4`. (`npx` just runs that command w/o installing it; and that version 9.4.4 is related to the current version of electron, which may change in an upcoming Atom release)

When I'm working on a tree-sitter grammar for Atom and I need to install or update a new tree-sitter package, I often myself doing *both* `npm rebuild` and `electron-rebuild --version 9.4.4` and it usually works out the kinks.
